Tools/performance_tests/cori.py

```diff
@@ -0,0 +1,158 @@
+import os, copy
+
+from functions_perftest import test_element
+
+module_name = {'cpu': 'haswell.', 'knl': 'mic-knl.', 'gpu':'.'}
+
+def executable_name(compiler, architecture):
+    return 'perf_tests3d.' + compiler + \
+        '.' + module_name[architecture] + 'TPROF.MPI.OMP.ex'
+
+def get_config_command(compiler, architecture):
+    config_command = ''
+    config_command += 'module unload darshan;'
+    if architecture == 'knl':
+        if compiler == 'intel':
+            config_command += 'module unload PrgEnv-gnu;'
+            config_command += 'module load PrgEnv-intel;'
+        elif compiler == 'gnu':
+            config_command += 'module unload PrgEnv-intel;'
+            config_command += 'module load PrgEnv-gnu;'
+        config_command += 'module unload craype-haswell;'
+        config_command += 'module load craype-mic-knl;'
+    elif architecture == 'cpu':
+        if compiler == 'intel':
+            config_command += 'module unload PrgEnv-gnu;'
+            config_command += 'module load PrgEnv-intel;'
+        elif compiler == 'gnu':
+            config_command += 'module unload PrgEnv-intel;'
+            config_command += 'module load PrgEnv-gnu;'
+        config_command += 'module unload craype-mic-knl;'
+        config_command += 'module load craype-haswell;'
+    return config_command
+
+# This function runs a batch script with 
+# dependencies to perform the analysis 
+# after all performance tests are done.
+def process_analysis(automated, cwd, compiler, architecture, n_node_list, start_date):
+    dependencies = ''
+    f_log = open(cwd + 'log_jobids_tmp.txt' ,'r')
+    for line in f_log.readlines():
+        dependencies += line.split()[3] + ':'
+    batch_string = ''
+    batch_string += '#!/bin/bash\n'
```

Comment:
If you like, you can write easier maintainable multiline strings in python:
https://stackoverflow.com/a/10660443/2719194

Reply:
Thanks, I used triple quotes where it clarified the code.